connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockPlayerInputTranslator.java

```diff
@@ -44,5 +52,27 @@ public void translate(PlayerInputPacket packet, GeyserSession session) {
         );
 
         session.sendDownstreamPacket(clientSteerVehiclePacket);
+
+        // Bedrock only sends movement vehicle packets while moving
+        // This allows horses to take damage while standing on magma
+        Entity vehicle = session.getRidingVehicleEntity();
+        if (vehicle instanceof BoatEntity || (vehicle instanceof AbstractHorseEntity && !(vehicle instanceof LlamaEntity))) {
```

Comment:
Why exclude the llamas?

Reply:
Ah, right, llamas aren't controllable. Does this need to be done for minecarts?

Reply:
I've just checked minecarts and Java clients don't send vehicle move packets at all. I've also fixed it so that players riding on the back of boats don't send vehicle move packets.